vm/src/util.rs

```diff
@@ -11,3 +12,20 @@ pub fn read_file(path: &Path) -> Result<String> {
 
     Ok(buffer)
 }
+
+pub(crate) fn atomic_iter_advance(
+    pos: &AtomicCell<usize>,
+    done: impl Fn(usize) -> bool,
+    next: impl Fn(usize) -> usize,
+) -> Option<usize> {
+    let mut prev = pos.load();
```

Comment:
Please see comment in `objiter`

Reply:
I made a comment on your last set of reviews; if it's a reverse iterator and it's going backward, then `fetch_sub` would underflow and the index would be `usize::MAX_VALUE`. What would you suggest for that?

Reply:
Maybe a separate `done` flag?

Reply:
I do prefer to keep this simple as the current implementation is a little complex. I think we an go in a few directions here:
1. Change the position to `isize`.
2. Start the position at `len(iterator)` instead of `len(iterator) - 1` and go until 0.
3. Check if we are at 0 before doing the `fetch_sub`.

Reply:
Oh, right, it'd be fine if we do a load and then a fetch_sub after we check that we're still in bounds, since it's still atomic. Okay, thanks for your feedback.

Reply:
Looking at this again I am not sure  2 or 3 will be simple at all. I think changing to `isize` will be the simplest. You can't actually do `load` and then `fetch_sub` as it will not be atomic.

Reply:
Okay, I updated it to just use the `fetch_*` operations. I didn't really want to do that, but I think I was just being silly, because yes, if someone does this:
```py
it = iter([1,2])
try:
    next(it)
except:
    pass
```
Then it's eventually going to wrap around and start yielding elements again, but that's a very small edge case that doesn't really matter in the grand scheme of things.